server/odc-service/src/main/java/com/oceanbase/odc/service/flow/FlowInstanceService.java

```diff
@@ -970,6 +1076,69 @@ private FlowInstanceConfigurer buildConfigurer(
+        List<FlowInstanceConfigurer> configurers = new LinkedList<>();
+        for (int nodeSequence = 0; nodeSequence < nodeConfigs.size(); nodeSequence++) {
+            FlowInstanceConfigurer configurer;
+            ApprovalNodeConfig nodeConfig = nodeConfigs.get(nodeSequence);
+            Long resourceRoleId = nodeConfig.getResourceRoleId();
+            FlowApprovalInstance approvalInstance = flowFactory.generateFlowApprovalInstance(flowInstance.getId(),
+                    false, false,
+                    nodeConfig.getAutoApproval(), approvalFlowConfig.getApprovalExpirationIntervalSeconds(),
+                    nodeConfig.getExternalApprovalId());
+            if (Objects.nonNull(resourceRoleId)) {
+                approvalInstance.setCandidate(StringUtils.join(flowInstanceReq.getProjectId(), ":", resourceRoleId));
+            }
+            FlowGatewayInstance approvalGatewayInstance =
+                    flowFactory.generateFlowGatewayInstance(flowInstance.getId(), false, true);
+            configurer = flowInstance.newFlowInstanceConfigurer(approvalInstance);
+            configurer = configurer.next(approvalGatewayInstance).route(String.format("${!%s}",
+                    FlowApprovalInstance.APPROVAL_VARIABLE_NAME), flowInstance.endFlowInstance());
+            if (nodeSequence == nodeConfigs.size() - 1) {
+                FlowInstanceConfigurer taskConfigurer = null;
+                int orders = ((MultipleDatabaseChangeParameters) flowInstanceReq.getParameters())
+                        .getOrderedDatabaseIds().size();
+                for (int i = 0; i < orders; i++) {
+                    ExecutionStrategyConfig strategyConfig = ExecutionStrategyConfig.from(flowInstanceReq,
+                            approvalFlowConfig.getWaitExecutionExpirationIntervalSeconds());
+                    FlowTaskInstance taskInstance;
+                    if (i == orders - 1) {
+                        taskInstance = flowFactory.generateFlowTaskInstance(flowInstance.getId(), false, true,
+                                taskType, strategyConfig);
+                    } else {
+                        taskInstance = flowFactory.generateFlowTaskInstance(flowInstance.getId(), false, false,
+                                taskType, strategyConfig);
+                    }
+
+                    taskInstance.setTargetTaskId(taskId);
+                    if (taskConfigurer == null) {
+                        taskConfigurer = flowInstance.newFlowInstanceConfigurer(taskInstance);
+                    } else {
+                        taskConfigurer.next(taskInstance);
+                    }
+                }
+                taskConfigurer.endFlowInstance();
+                configurer.route(String.format("${%s}", FlowApprovalInstance.APPROVAL_VARIABLE_NAME),
+                        taskConfigurer);
+            }
+            configurers.add(configurer);
+        }
+        for (int j = configurers.size() - 2; j >= 0; j--) {
```

Comment:
why `configurers.size() - 2` ?
does `configurers.size() - 2` always gereater than 0 ?

Reply:
we have 4 risk levels, and the approval node is required to be 1-2. When the number of approval nodes is 1, it will not enter the loop. When the number of approval nodes is 2, it will enter the loop